crates/sui-core/src/overload_monitor.rs

```diff
@@ -289,4 +390,264 @@ mod tests {
+                        authority.metrics.execution_rate_tracker.lock().record();
+                        authority.metrics.execution_queueing_latency.report(start_time.elapsed());
+                        interval.tick().await;
+                    }
+                    _ = &mut stop_rx => {
+                        info!("Executor stopping");
+                        return;
+                    }
+                }
+            }
+        })
+    }
+
+    // Helper fundtion to periodically print the current overload info.
+    async fn sleep_and_print_stats(state: Arc<AuthorityState>, seconds: u32) {
+        for _ in 0..seconds {
+            info!(
+                "Overload: {:?}. Shedding percentage: {:?}. Queue: {:?}, Ready rate: {:?}. Exec rate: {:?}.",
+                state.overload_info.is_overload.load(Ordering::Relaxed),
+                state
+                    .overload_info
+                    .load_shedding_percentage
+                    .load(Ordering::Relaxed),
+                state.metrics.execution_queueing_latency.latency(),
+                state.metrics.txn_ready_rate_tracker.lock().rate(),
+                state.metrics.execution_rate_tracker.lock().rate(),
+            );
+            sleep(Duration::from_secs(1)).await;
+        }
+    }
+
+    // Running a workload with consistent steady `generator_rate` and `executor_rate`.
+    // It checks that the dropped requests should in between min_dropping_rate and max_dropping_rate.
+    async fn run_consistent_workload_test(
+        generator_rate: f64,
+        executor_rate: f64,
+        min_dropping_rate: f64,
+        max_dropping_rate: f64,
+    ) {
+        let state = TestAuthorityBuilder::new().build().await;
+
+        let (tx, rx) = unbounded_channel();
+        let (_burst_tx, burst_rx) = unbounded_channel();
+        let total_requests = Arc::new(AtomicU32::new(0));
+        let dropped_requests = Arc::new(AtomicU32::new(0));
+        let load_generator = start_load_generator(
+            generator_rate,
+            tx.clone(),
+            burst_rx,
+            state.clone(),
+            true,
+            total_requests.clone(),
+            dropped_requests.clone(),
+        );
+
+        let (stop_tx, stop_rx) = oneshot::channel();
+        let executor = start_executor(executor_rate, rx, stop_rx, state.clone());
+
+        sleep_and_print_stats(state.clone(), 50).await;
+
+        stop_tx.send(()).unwrap();
+        let _ = tokio::join!(load_generator, executor);
+
+        let dropped_ratio = dropped_requests.load(Ordering::SeqCst) as f64
+            / total_requests.load(Ordering::SeqCst) as f64;
+        assert!(min_dropping_rate <= dropped_ratio);
+        assert!(dropped_ratio <= max_dropping_rate);
+    }
+
+    // Tests that when request generation rate is slower than execution rate, no requests should be dropped.
+    #[tokio::test(flavor = "multi_thread")]
+    #[cfg_attr(msim, ignore)]
```

Comment:
tokio::test tests are already ignored automatically in simtest builds so this should not be necessary